lib/src/http/http_provider.dart

```diff
@@ -210,12 +207,19 @@ class HttpProvider extends Provider with FluriMixin {
     if (reqUri == null || reqUri.toString() == '') throw new StateError(
         'HttpProvider: Cannot send a request without a URI.');
 
+    // If this is a retry, there will be a previous request stored in the meta.
+    HttpContext previousAttempt =
+        meta.containsKey('previousAttempt') ? meta['previousAttempt'] : null;
```

Comment:
Yeah I'd rather get rid of that, too, but I couldn't think of a good way to pass that information through. Currently the public HttpProvider API doesn't deal with HttpContext at all (which I like), that's only used internally and for interceptors. I'll keep noodling on it